lupa/_lupa.pyx

```diff
@@ -654,17 +683,16 @@ cdef tuple _fix_args_kwargs(tuple args):
         return args, {}
 
     cdef _LuaTable table = <_LuaTable>arg
-    table.push_lua_object(table._state)
-    return unpack_lua_table(table._runtime)
+    with table._runtime.stack(1):
+        table.push_lua_object(table._state)
+        return unpack_lua_table(table._runtime)
 
 
 ################################################################################
 # fast, re-entrant runtime locking
 
-cdef inline int lock_runtime(LuaRuntime runtime) except -1:
-    if not lock_lock(runtime._lock, pythread.PyThread_get_thread_ident(), True):
-        raise LuaError("Failed to acquire thread lock")
-    return 0
+cdef inline bint lock_runtime(LuaRuntime runtime) with gil:
+    return lock_lock(runtime._lock, pythread.PyThread_get_thread_ident(), True)
```

Comment:
This used to raise an exception and it no longer does. Is that intended?

Reply:
(read your description, it _is_ intended)

Why is the `with gil` declaration necessary?